docs/source/migration-guides/post-0.8-deserialization.md

```diff
@@ -0,0 +1,264 @@
+# Post-0.8 deserialization API migration guide
+
+After 0.8, a new deserialization API has been introduced. The new API improves type safety and performance of the old one, so it is highly recommended to switch to it. However, deserialization is an area of the API that users frequently interact with: deserialization traits appear in generic code and custom implementations have been written. In order to make migration easier, the driver still offers the old API, which - while opt-in - can be very easily switched to after version upgrade. Furthermore, a number of facilities have been introduced which help migrate the user code to the new API piece-by-piece.
+
+The old API and migration facilities will be removed in the next major release (2.0).
+
+## Introduction 
+
+### Old traits
+
+The legacy API works by deserializing rows in the query response to a sequence of `Row`s. The `Row` is just a `Vec<Option<CqlValue>>`, where `CqlValue` is an enum that is able to represent any CQL value.
+
+The user can request this type-erased representation to be converted into something useful. There are two traits that power this:
+
+__`FromRow`__
+
+```rust
+# extern crate scylla;
+# use scylla::frame::response::cql_to_rust::FromRowError;
+# use scylla::frame::response::result::Row;
+pub trait FromRow: Sized {
+    fn from_row(row: Row) -> Result<Self, FromRowError>;
+}
+```
+
+__`FromCqlVal`__
+
+```rust
+# extern crate scylla;
+# use scylla::frame::response::cql_to_rust::FromCqlValError;
+// The `T` parameter is supposed to be either `CqlValue` or `Option<CqlVal>`
+pub trait FromCqlVal<T>: Sized {
+    fn from_cql(cql_val: T) -> Result<Self, FromCqlValError>;
+}
+```
+
+These traits are implemented for some common types:
+
+- `FromRow` is implemented for tuples up to 16 elements,
+- `FromCqlVal` is implemented for a bunch of types, and each CQL type can be converted to one of them.
+
+While it's possible to implement those manually, the driver provides procedural macros for automatic derivation in some cases:
+
+- `FromRow` - implements `FromRow` for a struct.
+- `FromUserType` - generated an implementation of `FromCqlVal` for the struct, trying to parse the CQL value as a UDT.
+
+Note: the macros above have a default behavior that is different than what `FromRow` and `FromUserTypes` do.
+
```

Comment:
This is unclear.

Reply:
What is their default behaviour and what are the differences? Also, as the names here are similar both for macros and traits, it is crucial to distinguish them with some strong markers.